src/transformers/generation/logits_process.py

```diff
@@ -106,12 +107,12 @@ class MinLengthLogitsProcessor(LogitsProcessor):
 
     def __init__(self, min_length: int, eos_token_id: Union[int, List[int]]):
         if not isinstance(min_length, int) or min_length < 0:
-            raise ValueError(f"`min_length` has to be a positive integer, but is {min_length}")
+            warnings.warn(f"`min_length` has to be a positive integer, but is {min_length}")
```

Comment:
Here what would be the goal of setting a `min_length` to a negative value? (nit: the error message should say non-negative by the way).

Reply:
Ah yes, I thought this was needed to enable open-ended generation but I just checked and it's not the case. Now reverted back to raising an error.